src/algorithms/PVT/libs/rtcm_printer.cc

```diff
@@ -240,1270 +242,146 @@ void Rtcm_Printer::Print_Rtcm_Messages(const Rtklib_Solver* pvt_solver,
-                            break;
-                        case 107:
-                            if (rtcm_MT1019_rate_ms != 0)  // allows deactivating messages by setting rate = 0
+                        }
+                    else if (rtcm_MT1087_enabled && only_glonass)
+                        {
+                            const auto glo_gnav_ephemeris_iter = pvt_solver->glonass_gnav_ephemeris_map.cbegin();
+                            if (glo_gnav_ephemeris_iter != pvt_solver->glonass_gnav_ephemeris_map.cend())
                                 {
-                                    for (const auto& gps_eph_iter : pvt_solver->gps_ephemeris_map)
-                                        {
-                                            Print_Rtcm_MT1019(gps_eph_iter.second);
-                                        }
+                                    Print_Rtcm_MSM(7, {}, {}, {}, glo_gnav_ephemeris_iter->second, rx_time, gnss_observables_map, enable_rx_clock_correction, 0, 0, false, false);
                                 }
-                            if (rtcm_MSM_rate_ms != 0)
-                                {
-                                    auto gps_eph_iter = pvt_solver->gps_ephemeris_map.cbegin();
-                                    int gps_channel = 0;
-                                    for (const auto& gnss_observables_iter : gnss_observables_map)
-                                        {
-                                            const std::string system(gnss_observables_iter.second.System, 1);
-                                            if (gps_channel == 0)
-                                                {
-                                                    if (system == "G")
-                                                        {
-                                                            // This is a channel with valid GPS signal
-                                                            gps_eph_iter = pvt_solver->gps_ephemeris_map.find(gnss_observables_iter.second.PRN);
-                                                            if (gps_eph_iter != pvt_solver->gps_ephemeris_map.cend())
-                                                                {
-                                                                    gps_channel = 1;
-                                                                }
-                                                        }
-                                                }
-                                        }
-                                    if (gps_eph_iter != pvt_solver->gps_ephemeris_map.cend())
-                                        {
-                                            Print_Rtcm_MSM(7, gps_eph_iter->second, {}, {}, {}, rx_time, gnss_observables_map, enable_rx_clock_correction, 0, 0, false, false);
-                                        }
+                        }
+                    else
```

Comment:
For MSM, I tried to keep the same logic as before, but I am a bit confused why it's necessary.

Why can't we simply take the first ephemeris of each map and use that, instead of going in the gnss observables map and then looking into the ephemeris maps and take the first that match? For the previous cases with one cosntellation, we only looked into the ephemeris maps, so I really don't see the reason to do this.